### PR TITLE
Added code review checklist

### LINKED ISSUE #345: Code review checklist
Creating a list that eventually will turn into some type of documentation for commonly identified items that might need to be checked for in a code review

Code Conventions
- Check ambiguous comparisons with numpy arrays, switching them to if y is not None etc, aka explicit testing for 'truthy' / 'falsy' values that might be obscured with numpy and other data containers
- added random_state to the visualizer to engage image comparison tests
- user np.unique() instead of set()
- self.classes should be self.classes_ (same for self.score_)

Testing
- Ensure that there is an image comparison test for visualizers
- A random_state to the visualizer to engage image comparison tests
- Ensure there are tests for a variety of inputs formats included pandas dataframes, numpy arrays
- Ensure that fit returns self

Doc String
- Visualizer doc string is under the class
- Contains narrative about visualizer and parameters with correct formatting

### PR DESCRIPTION
**Pull request by bbengfort**
Broke down the contributing.rst file into multiple smaller files for
easier editing. In the advanced development topics section I added a
section for common code, testing, and documentation conventions, listing
some things that Nathan mentioned and that are part of the ongoing
visualizer audit process.

Fixes #345

### CODE REVIEW

**Comment by bbengfort** [comment]
@ndanielsen mind doing a review? The only new content is in the "Visualizer Review Checklist" section in ``advanced_development_topics.rst`` -- the rest of it is just breaking down our ``contributing.rst`` into smaller, more easily edited chunks.

**Comment by ndanielsen** [comment]
Sure, will take a look in the next day or so. =)

File: docs/contributing/advanced_development_topics.rst [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+- Fix all ``random_state`` arguments.
+
+    Be on the lookout for any method (particularly sklearn methods) that have a ``random_state`` argument and be sure to fix them so that tests always pass!
+
+- Test a variety of inputs.
+
+    Machine learning can be done on a variety of inputs for ``X`` and ``y``, ensure there is a test with numpy arrays, pandas DataFrame and Series objects, and with Python lists.
+
+- Test that ``fit()`` returns self.
```
Comment (ndanielsen) [inline docs/contributing/advanced_development_topics.rst]: I feel like we should implement a test helper function for this.

File: docs/contributing/advanced_development_topics.rst [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    When doing end-to-end testing, we like to ``assert oz.fit() is oz`` to ensure the API is maintained.
+
+- Test that ``score()`` between zero and one.
+
+    With visualizers that have a ``score()`` method we like to ``assert 0.0 <= oz.score() >=1.0`` to ensure the API is maintained.
```
Comment (ndanielsen) [inline docs/contributing/advanced_development_topics.rst]: I think this needs a comma, `method, we like to`

File: docs/contributing/advanced_development_topics.rst [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+Head back to Waffle and checkout another issue!
+
+Releases
+--------
+
+When ready to create a new release we branch off of develop as follows::
+
+    $ git checkout -b release-x.x
+
+This creates a release branch for version x.x. At this point do the version bump by modifying ``version.py`` and the test version in ``tests/__init__.py``. Make sure all tests pass for the release and that the documentation is up to date. There may be style changes or deployment options that have to be done at this phase in the release branch. At this phase you'll also modify the ``changelog`` with the features and changes in the release.
```
Comment (ndanielsen) [inline docs/contributing/advanced_development_topics.rst]: Do we have instructions somewhere for building the documentation that we can reference here? Feel like it's a big thing to make sure that docs are built and working before the release

File: docs/contributing/advanced_development_topics.rst [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+Advanced Development Topics
+===========================
+
+In this section we discuss more advanced contributing guidelines such as code conventions,the release life cycle or branch management. This section is intended for maintainers and core contributors of the Yellowbrick project. If you would like to be a maintainer please contact one of the current maintainers of the project.
+
+Visualizer Review Checklist
+---------------------------
+
+As the visualizer API has matured over time, we've realized that there are a number of routine items that must be in place to consider a visualizer truly complete and ready for prime time. This list is also extremely helpful for reviewing code submissions to ensure that visualizers are consistently implemented, tested, and documented. Though we do not expect these items to be checked off on every PR, the below list includes some guidance about what to look for when reviewing or writing a new Visualizer.
```
Comment (ndanielsen) [inline docs/contributing/advanced_development_topics.rst]: Perhaps add a sentence about `contrib` as a great place for work in progress visualizers.

File: docs/contributing/advanced_development_topics.rst [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+Code Conventions
```
Comment (ndanielsen) [inline docs/contributing/advanced_development_topics.rst]: These are a great collection of learnings - almost worthy of a pydata talk =).  

One thing that I would add is trying to keep the API organized where it makes sense, ie matplotlib in `draw`, scikit-learn model fitting  in `fit()`, etc. I think this captured in other docs but a little repetition here might be helpful for newer reviewers

**Comment by ndanielsen** [comment]
@bbengfort just took a look. In general, the code review checklist is very solid. Just a couple minor points!

File: docs/contributing/advanced_development_topics.rst (hunk 1 above)
Comment (bbengfort) [reply docs/contributing/advanced_development_topics.rst]: We have the checks module framework from #180 that was supposed to do this for us, but we didn't get too far on it.

File: docs/contributing/advanced_development_topics.rst (hunk 4 above)
Comment (bbengfort) [reply docs/contributing/advanced_development_topics.rst]: Add a note block below this paragraph.

File: docs/contributing/advanced_development_topics.rst (hunk 5 above)
Comment (bbengfort) [reply docs/contributing/advanced_development_topics.rst]: Added it to the list of conventions!

**Pull request merged**